core/trino-main/src/test/java/io/trino/operator/output/TestPositionsAppender.java

```diff
@@ -215,6 +216,21 @@ public void testConsecutiveBuilds(Type type)
         assertEquals(positionsAppender.build().getPositionCount(), 0);
     }
 
+    // testcase for jit bug described https://github.com/trinodb/trino/issues/12821
+    @Test(priority = Integer.MIN_VALUE)
```

Comment:
Could you add a comment why `priority = Integer.MIN_VALUE` is needed ?

Reply:
good point, comment added.

Reply:
How do you know that other test executions (in other classes) don't affect jit compilation. Should you use isolation instead?

Reply:
I did test it manually and `mvn test` in the trino-main triggers this issue.
I think class isolation for this is an overkill

Reply:
Was there any query which was reproducing it consistently ?
Would just creating a separate test class for this be sufficient ?

Reply:
> Was there any query which was reproducing it consistently ?

There was one on hive, at least on sf10 (https://github.com/trinodb/trino/issues/12821#issuecomment-1156107111).
It doesn't trigger the bug on tpch.sf1;

> Would just creating a separate test class for this be sufficient ?

I'm not sure adding a new test class is simpler than adding a test method. Especially because it would be dependent on this test class